src/idl_gen_csharp.cpp

```diff
@@ -1266,19 +1266,29 @@ class CSharpGenerator : public BaseGenerator {
     code += "  }\n\n";
     // As<T>
     code += "  public T As<T>() where T : class { return this.Value as T; }\n";
-    // As
+    // As, From
     for (auto it = enum_def.Vals().begin(); it != enum_def.Vals().end(); ++it) {
       auto &ev = **it;
       if (ev.union_type.base_type == BASE_TYPE_NONE) continue;
       auto type_name = GenTypeGet_ObjectAPI(ev.union_type, opts);
-      if (ev.union_type.base_type == BASE_TYPE_STRUCT &&
-          ev.union_type.struct_def->attributes.Lookup("private")) {
-        code += "  internal ";
-      } else {
-        code += "  public ";
+      std::string accessibility =
+          (ev.union_type.base_type == BASE_TYPE_STRUCT &&
+           ev.union_type.struct_def->attributes.Lookup("private"))
+              ? "internal"
+              : "public";
+      // As
+      code += "  " + accessibility + " " + type_name + " As" + ev.name +
+              "() { return this.As<" + type_name + ">(); }\n";
+      // From
+      std::string lower_ev_name;
+      for (auto name_it = ev.name.begin(); name_it != ev.name.end();
```

Comment:
Probably this loop can be rewritten with `std::transform`:
```cpp
auto lower_ev_name = ev.name; // make copy
std::transform(lower_ev_name.begin(), lower_ev_name.end(), lower_ev_name.begin(), CharToLower);
```
or 
```cpp
std::string lower_ev_name;
std::transform(ev.name.begin(), ev.name..end(), std::back_inserter(lower_ev_name), CharToLower);
```

Reply:
@vglavnyy fixed it.